### PR TITLE
Adds Typescript support to React

### PR DESCRIPTION
**Pull request by oicramps**
# Summary

- Adds typescript support to the React part of the BRModelo app;
- Configures `styled-component` theme;
- Creates a providers wrapper;

### CODE REVIEW

**Comment by idmarjr** [comment]
@oicramps I would like to have session to walk through those changes with you so I can have a better understanding of what's happening. :)

**Comment by oicramps** [comment]
> @oicramps I would like to have session to walk through those changes with you so I can have a better understanding of what's happening. :)

Yeah, for sure!

File: app/react/theme/index.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+		gray: {
+			100: "#f8f8f8",
+		},
+		white: "#fff",
+		black: "#000",
+	},
+	shadows: {
+		default: "0 0 20px rgb(0, 0, 0, 0.1)",
+	},
+	radii: {
```
Comment (ArthurMota9) [inline app/react/theme/index.ts:18]: Radii? I didn't understand this part 😞

**Comment by ArthurMota9** [comment]
Good job, my friend! Your contributions are always amazing! 🚀 ❤️

File: app/react/theme/index.ts (hunk 1 above)
Comment (oicramps) [reply app/react/theme/index.ts:18]: It was created to define `border-radius` in our theme. We'll start the migration with the Login page, so I already defined it to use in the `Card` component. But it is all opened to discussion, mainly this theme definition.

**Comment by oicramps** [comment]
image

**Comment by idmarjr** [comment]
We did a round of tests headless and interactive and everything works fine.
@wlsf82 Can you double check?

**Review by idmarjr — APPROVED** [review]

**Review by ArthurMota9 — APPROVED** [review]

**Review by miltonbsn — APPROVED** [review]

**Comment by wlsf82** [comment]
> We did a round of tests headless and interactive and everything works fine. @wlsf82 Can you double-check?

All tests passed here also, both in headful and headless mode.

Still, the test that creates a logical model using drag-and-drop acts differently in headless mode (I'll adjust that in another PR.)

**Pull request merged**